From 9969408d221befe00c9d392627f95b1fbde79ff8 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Wed, 11 Sep 2019 20:18:24 +0530
Subject: [PATCH 06/15] octeontx2-pf: Do not set mac address again

commit c36a4a18c6d9cba7dbc59cd64cec4a00187918a6 from
git@git.assembla.com:cavium/WindRiver.linux.git

AF driver sets mac address for all PFs/VFs.
Setting MAC address again in otx2_open enables
RX traffic before setting up NAPI hence removed
it.

Change-Id: Icd6182d4428b81100c5d3d55dcddd2f2da870caa
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15665
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/nic/otx2_pf.c    | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index e87216d2a1e0..3cd127cc0f65 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1517,11 +1517,6 @@ int otx2_open(struct net_device *netdev)
 		napi_enable(&cq_poll->napi);
 	}
 
-	/* Set default mac address */
-	err = otx2_hw_set_mac_addr(pf, netdev);
-	if (err)
-		goto err_disable_napi;
-
 	/* Set default MTU in HW */
 	err = otx2_hw_set_mtu(pf, netdev->mtu);
 	if (err)
@@ -1582,10 +1577,6 @@ int otx2_open(struct net_device *netdev)
 	/* 'intf_down' may be checked on any cpu */
 	smp_wmb();
 
-	err = otx2_rxtx_enable(pf, true);
-	if (err)
-		goto err_free_cints;
-
 	/* we have already received link status notification */
 	if (pf->linfo.link_up && !(pf->pcifunc & RVU_PFVF_FUNC_MASK))
 		otx2_handle_link_event(pf);
@@ -1600,6 +1591,10 @@ int otx2_open(struct net_device *netdev)
 		otx2_config_hw_rx_tstamp(pf, true);
 	}
 
+	err = otx2_rxtx_enable(pf, true);
+	if (err)
+		goto err_free_cints;
+
 	return 0;
 
 err_free_cints:
-- 
2.17.1

